pkg/operator/aws_platform_service_location/controller.go

```diff
@@ -38,12 +49,92 @@ func NewController(operatorClient operatorv1helpers.OperatorClient, infraClient
 }
 
 func (c AWSPlatformServiceLocationController) sync(ctx context.Context, syncCtx factory.SyncContext) error {
-	currentInfra, err := c.infraClient.Infrastructures().Get(ctx, "cluster", metav1.GetOptions{})
+	obj, err := c.infraLister.Get("cluster")
 	if errors.IsNotFound(err) {
-		// log not found errors, but don't degrade?
-		klog.Warningf("%w", err)
+		syncCtx.Recorder().Warningf("AWSPlatformServiceLocationController", "Required infrastructures.%s/cluster not found", configv1.GroupName)
 		return nil
 	}
-	klog.Info(spew.Sprintf("current state: %s", currentInfra.Status))
+	if err != nil {
+		return err
+	}
+
+	currentInfra := obj.DeepCopy()
+	var platformName configv1.PlatformType
+	if pstatus := currentInfra.Status.PlatformStatus; pstatus != nil {
+		platformName = pstatus.Type
+	}
+	if len(platformName) == 0 {
+		syncCtx.Recorder().Warningf("AWSPlatformServiceLocationController", "Falling back to deprecated status.platform because infrastructures.%s/cluster status.platformStatus.type is empty", configv1.GroupName)
+		platformName = currentInfra.Status.Platform
+	}
+
+	if platformName != configv1.AWSPlatformType {
+		return nil // nothing to do here.
+	}
+
```

Comment:
if the spec.platformspec.type != "" && spec.platformspec.type != platformName, it should error.